### ISSUE TITLE
[MJAVADOC-411] Package does not exist warnings

### ISSUE THREAD

**Issue by jira-importer** [apache/maven-javadoc-plugin#702]
**[Anthony Whitford](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=awhitford)** opened **[MJAVADOC-411](https://issues.apache.org/jira/browse/MJAVADOC-411?redirect=false)** and commented

Since version 2.10, it appears that Javadoc is warning about missing packages when generating javadoc.

* For the `javadoc` goal, the `compile` scope classes should be in scope
* For the `test-javadoc` goal, the `test` scope classes should be in scope

---

**Affects:** 2.10

**Issue Links:**
- [MJAVADOC-406](https://issues.apache.org/jira/browse/MJAVADOC-406) Regression: MJAVADOC-398 commit changed wrong line
 (_**"duplicates"**_)
- [MJAVADOC-414](https://issues.apache.org/jira/browse/MJAVADOC-414) Artifacts missing from test classpath.

2 votes, 3 watchers

**Comment by jira-importer** [comment]
**[Dennis Lundberg](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=dennisl@apache.org)** commented

We need more info to be able to understand your problem.

Could your problem be the same as MJAVADOC-407?

**Comment by jira-importer** [comment]
**[Anthony Whitford](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=awhitford)** commented

While I suspect the issue is a side effect of MJAVADOC-398, I don't think it is a duplicate of MJAVADOC-407.

When building, I am now getting warnings such as these.  For example:

```
[WARNING] Javadoc Warnings
[WARNING] /Users/anthony/Documents/lombok.maven/test-maven-lombok/src/test/java/org/projectlombok/test/AppTest.java:3: error: package junit.framework does not exist
[WARNING] import junit.framework.Test;
[WARNING] ^
[WARNING] /Users/anthony/Documents/lombok.maven/test-maven-lombok/src/test/java/org/projectlombok/test/AppTest.java:4: error: package junit.framework does not exist
[WARNING] import junit.framework.TestCase;
[WARNING] ^
[WARNING] /Users/anthony/Documents/lombok.maven/test-maven-lombok/src/test/java/org/projectlombok/test/AppTest.java:5: error: package junit.framework does not exist
[WARNING] import junit.framework.TestSuite;
[WARNING] ^
[WARNING] /Users/anthony/Documents/lombok.maven/test-maven-lombok/src/test/java/org/projectlombok/test/AppTest.java:10: error: cannot find symbol
[WARNING] public class AppTest extends TestCase
[WARNING] ^
[WARNING] symbol: class TestCase
[WARNING] /Users/anthony/Documents/lombok.maven/test-maven-lombok/src/test/java/org/projectlombok/test/AppTest.java:24: error: cannot find symbol
[WARNING] public static Test suite() {
```

The above is from the `test-javadoc` goal, and it is complaining that it can't find JUnit classes, yet JUnit is `test` dependency.

**Comment by jira-importer** [comment]
**[Anthony Whitford](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=awhitford)** commented

I suspect that MJAVADOC-406 (which is essentially applying the proper fix to MJAVADOC-398) will solve the issue.

**Comment by jira-importer** [comment]
**[Andrei Khveras](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=tenety)** commented

This issue affects us as well.
Once maven-javadoc-plugin v. 2.10 released in the end of september, we got unable to build our project due to failing maven goal:

`Failed to execute goal org.apache.maven.plugins:maven-javadoc-plugin:2.10:jar (attach-javadocs) on project framework-core: MavenReportException: Error while creating archive:
[ERROR] Exit code: 1 - /mnt/jenkins/jobs/Release Framework/workspace/framework-core/src/main/java/framework/core/assertions/DeferredAsserterListener.java:3: package org.testng does not exist`

This is definitely not an issue in our code as once I did a trick replacing maven-javadoc-plugin v.2.10 to v.2.9.1 in local Maven cache, I got able to build our framework as usual.

Environments: Linux, Java 1.6, Maven 2.2.1; Windows 8.1, Java 1.6, Maven 3.2.3
